src/shogun/base/progress.h

```diff
@@ -58,6 +58,10 @@
 
 namespace shogun
 {
+//#define progress(...)
+// Progress(std::string(this->get_name())+"::"+std::string(__FUNCTION__),
```

Comment:
due to a class that use a progress bar being external i have commented it out for now

Reply:
not sure I understand

Reply:
I think that is caused by the fact that some Shogun's class does not have the `get_name` method, because they do not inherith from SGObject. For those classes I think that using the standard progress bar could be enough.

Reply:
lets have PROGRESS and SPROGRESS (static version). The static version accepts a string as the first argument, callers have to provide the class name manually in those cases